app/controllers/cases/filters_controller.rb

```diff
@@ -133,6 +135,29 @@ def open
 
     private
 
+    def set_filtered_tab_counts
+      @global_nav_manager.current_page.tabs.each do |tab| 
+        tab.set_count(
+          filtered_count_for_tab(
+            cases_for_tab(tab)
+          )
+        )
+      end
+    end
+
+    def filtered_count_for_tab(tab_case_list)
+      call_search_service(tab_case_list).result_set.count
+    end
+
+    def cases_for_tab(tab)
+      tab.cases.includes(
+        :message_transitions,
+        :responder,
+        :responding_team,
+        :approver_assignments
+      )
+    end
```

Comment:
It is good idea to introduce an attribute to store the count for tab. 
If just for getting the number of cases,  there is no need to pre-load all those sub tables like `message_transitions, responders etc. `

you can just use `tab.cases` directly,   the cases_for_tab may not be needed.